packages/jest-config/src/index.ts

```diff
@@ -313,9 +307,21 @@ export function readConfigs(
 
         return true;
       })
-      .map((root, projectIndex) =>
-        readConfig(argv, root, true, configPath, projectIndex),
-      );
+      .map((root, projectIndex) => {
+        // Skip on all except: projectIndex === 0, and isGlobalProject, and projects.length === 1
+        const skipArgvConfigOption = !(
+          projectIndex === 0 &&
+          isGlobalProject &&
+          projects.length === 1
+        );
```

Comment:
The issue was that `skipArgvConfigOption` should only be `true` if the project is not "global". By global, I mean, it is not added from `getProjectListFromCLIArgs` using `process.cwd`